### PR TITLE
[KOGITO-3914] Incorporate Serverless Workflow Java SDK

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <zanini@redhat.com>

See: https://issues.redhat.com/browse/KOGITO-3914

In this PR we introduce the SW Java SDK to Kogito Core. This will make things farly easier for us to stay up to date with the specification. Since we don't have the `type` attribute for functions anymore, we are using `metadata.type`. This should vanish after [KOGITO-3915](https://issues.redhat.com/browse/KOGITO-3915) implementation, where we are going to support only the REST Calls implementation through OpenAPI definitions.

I also removed Camel, Decisions and Rules functions since that won't make sense anymore. All these services should be accessed via REST. The examples parsing test also was removed since this test is now being handled by the SDK itself.

`script`, `service`, `sysout` should also vanish in the near future (TODO: open a JIRA to follow this up).

I'll update the examples and test each of them before asking for review.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by ricardozanini** [comment]
Depends on: https://github.com/serverlessworkflow/sdk-java/pull/38

**Comment by ricardozanini** [comment]
Waiting for SW SDK 1.0.3 to be released.

**Comment by tsurdilo** [comment]
> Waiting for SW SDK 1.0.3 to be released.

It is released :) https://search.maven.org/search?q=io.serverlessworkflow

**Comment by ricardozanini** [comment]
I'm working on the updates! \o/

**Comment by ricardozanini** [comment]
@tsurdilo ca you please take a look?

File: jbpm/jbpm-serverless-workflow/src/test/resources/exec/eventbased-switch-state.sw.yml [hunk 1]
```diff
@@ -15,8 +15,7 @@ events:
 states:
   - name: AddRandomInfo
     type: inject
-    start:
-      kind: default
+    start: true
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply jbpm/jbpm-serverless-workflow/src/test/resources/exec/eventbased-switch-state.sw.yml:18]: Yes!

File: kogito-build-parent/pom.xml [hunk 2]
```diff
@@ -72,6 +72,7 @@
     <version.source.plugin>3.0.1</version.source.plugin>
     <version.surefire.plugin>2.22.2</version.surefire.plugin> <!-- minimum required by JUnit 5 -->
     <version.surefire.report.plugin>2.6</version.surefire.report.plugin>
+    <version.swagger.plugin>2.1.0</version.swagger.plugin>
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply kogito-build-parent/pom.xml:75]: The examples use it. @cristianonicolai removed, but we need it in the Spring examples. It was failing locally.

File: jbpm/jbpm-serverless-workflow/pom.xml [hunk 3]
```diff
@@ -52,32 +51,38 @@
...
+      <groupId>javax.validation</groupId>
+      <artifactId>validation-api</artifactId>
     </dependency>
+
     <dependency>
-      <groupId>com.fasterxml.jackson.dataformat</groupId>
-      <artifactId>jackson-dataformat-yaml</artifactId>
+      <groupId>io.serverlessworkflow</groupId>
+      <artifactId>serverlessworkflow-api</artifactId>
+      <version>${version.io.serverlessworkflow}</version>
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply jbpm/jbpm-serverless-workflow/pom.xml:67]: Oh, I missed that one.

File: jbpm/jbpm-serverless-workflow/src/test/resources/examples/applicantworkflow.sw.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
+{
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply jbpm/jbpm-serverless-workflow/src/test/resources/examples/applicantworkflow.sw.json:1]: In the test. I was validating the XORs with this example.

https://github.com/kiegroup/kogito-runtimes/pull/979/files#diff-253ff1ca5f759dfb68da68cc6bbcaf91aba3fa9783707f2d2122713695a40067R659

File: kogito-build-parent/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (danielezonca) [reply kogito-build-parent/pom.xml:75]: Mmm which example? Because with this PR
https://github.com/kiegroup/kogito-examples/pull/528
We migrated to a different dependency because swagger plugin was not working with sprint web annotations,
If we still have usages they are probably leftovers 🤔

[the remark this answers is not in the document]
Comment (ricardozanini) [reply kogito-build-parent/pom.xml:75]: Hmmm, I thought I had rebased the examples. It looks like I've missed that. My local branch still uses the plugin. Let me try again.

[the remark this answers is not in the document]
Comment (ricardozanini) [reply kogito-build-parent/pom.xml:75]: Yup, rebased and worked. I'll push an update to remove this.

**Comment by tsurdilo** [comment]
@ricardozanini looks good +1

**Review by tsurdilo — APPROVED** [review]

**Comment by ricardozanini** [comment]
Errors come from the OptaPlanner examples. 

Jenkins retest this please

File: jbpm/jbpm-serverless-workflow/src/main/java/org/jbpm/serverless/workflow/parser/core/ServerlessWorkflowFactory.java [hunk 5]
```diff
@@ -248,15 +247,15 @@ public void addMessageEndNodeAction(EndNode endNode, String variable, String mes
...
-                        "workItem.setName(\"Send Task\");" + EOL +
-                        "workItem.setNodeInstanceId(kcontext.getNodeInstance().getId());" + EOL +
-                        "workItem.setProcessInstanceId(kcontext.getProcessInstance().getId());" + EOL +
-                        "workItem.setNodeId(kcontext.getNodeInstance().getNodeId());" + EOL +
-                        "workItem.setParameter(\"MessageType\", \"" + messageType + "\");" + EOL +
-                        (variable == null ? "" : "workItem.setParameter(\"Message\", " + variable + ");" + EOL) +
-                        "workItem.setDeploymentId((String) kcontext.getKnowledgeRuntime().getEnvironment().get(\"deploymentId\"));" + EOL +
-                        "((org.drools.core.process.instance.WorkItemManager) kcontext.getKnowledgeRuntime().getWorkItemManager()).internalExecuteWorkItem(workItem);"));
+                                                "org.drools.core.process.instance.impl.WorkItemImpl workItem = new org.drools.core.process.instance.impl.WorkItemImpl();" + EOL +
```
Comment (evacchi) [inline jbpm/jbpm-serverless-workflow/src/main/java/org/jbpm/serverless/workflow/parser/core/ServerlessWorkflowFactory.java:250]: such long strings could be moved to templates

**Review by evacchi — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]

File: jbpm/jbpm-serverless-workflow/src/main/java/org/jbpm/serverless/workflow/parser/core/ServerlessWorkflowFactory.java (hunk 5 above)
Comment (ricardozanini) [reply jbpm/jbpm-serverless-workflow/src/main/java/org/jbpm/serverless/workflow/parser/core/ServerlessWorkflowFactory.java:250]: @evacchi, I believe we need to refactor the generation of these files. I opened [KOGITO-4304](https://issues.redhat.com/browse/KOGITO-4304) to follow this up.

**Comment by ricardozanini** [comment]
CI:
```
ERROR: Error accessing files to attach: /home/jenkins/workspace/KIE/kogito/ghprb-webhooks/kogito-runtimes does not exist.
Sending email to: noreply@github.com
```

Trying again...

Jenkins retest this please

**Comment by ricardozanini** [comment]
Jenkins retest this please

**Comment by ricardozanini** [comment]
@evacchi can be merged with https://github.com/kiegroup/kogito-examples/pull/518

CI is green! :hand_over_mouth:
